misc/scripts/install-local.sh

```diff
@@ -444,77 +443,78 @@ fi
+	# Check the integrity
+	git fsck --full
+else
+	case "$url" in
+		*.zip)
+			download "$url"
+			# hash the file
+			if ! hashcheck "${url##*/}"; then
+				return 1
 			fi
-
-			fancy_message info "Storing pacscript"
-			sudo mkdir -p /var/cache/pacstall/"$PACKAGE"/"$version"
-			if ! cd "$DIR" 2> /dev/null; then
-				error_log 1 "install $PACKAGE"; fancy_message error "Could not enter into ${DIR}"; exit 1
+			# unzip file
+			unzip -q "${url##*/}" 1>&1 2>/dev/null
+			# cd into it
+			cd ./*/ 2> /dev/null || { error_log 1 "install $PACKAGE"; fancy_message warn "Could not enter into the downloaded archive"; }
+		;;
+		*.deb)
+			download "$url"
+			if ! hashcheck "${url##*/}"; then
+				return 1
 			fi
-			sudo cp -r "$PACKAGE".pacscript /var/cache/pacstall/"$PACKAGE"/"$version"
-			sudo chmod o+r /var/cache/pacstall/"$PACKAGE"/"$version"/"$PACKAGE".pacscript
-			fancy_message info "Cleaning up"
-			cleanup
-			return 0
+			if sudo apt install -y -f ./"${url##*/}" 2>/dev/null; then
+				log
+				if type -t postinst > /dev/null 2>&1; then
+					if ! postinst; then
+						error_log 5 "postinst hook"
+						fancy_message error "Could not run postinst hook successfully"
+						exit 1
+					fi
+				fi
 
-		else
-			fancy_message error "Failed to install the package"
-			error_log 14 "install $PACKAGE"
-			sudo dpkg -r "$name" > /dev/null
-			fancy_message info "Cleaning up"
-			cleanup
-			return 1
-		fi
-	;;
-	*.AppImage)
-		download "$url"
-		if ! hashcheck "${url##*/}"; then
-			return 1
-		fi
-	;;
-	*)
-		download "$url"
-		# I think you get it by now
-		if ! hashcheck "${url##*/}"; then
-			return 1
-		fi
-		tar -xf "${url##*/}" 1>&1 2>/dev/null
-		cd ./*/ 2> /dev/null || ( error_log 1 "install $PACKAGE"; fancy_message warn "Could not enter into the downloaded archive" )
-	;;
-esac
+				fancy_message info "Storing pacscript"
+				sudo mkdir -p /var/cache/pacstall/"$PACKAGE"/"$version"
+				if ! cd "$DIR" 2> /dev/null; then
+					error_log 1 "install $PACKAGE"; fancy_message error "Could not enter into ${DIR}"; exit 1
+				fi
+				sudo cp -r "$PACKAGE".pacscript /var/cache/pacstall/"$PACKAGE"/"$version"
+				sudo chmod o+r /var/cache/pacstall/"$PACKAGE"/"$version"/"$PACKAGE".pacscript
+				fancy_message info "Cleaning up"
+				cleanup
+				return 0
+
+			else
+				fancy_message error "Failed to install the package"
+				error_log 14 "install $PACKAGE"
+				sudo dpkg -r "$name" > /dev/null
+				fancy_message info "Cleaning up"
+				cleanup
+				return 1
+			fi
+		;;
+		*.AppImage)
+			download "$url"
+			if ! hashcheck "${url##*/}"; then
+				return 1
+			fi
+		;;
+		*)
+			download "$url"
+			# I think you get it by now
+			if ! hashcheck "${url##*/}"; then
+				return 1
+			fi
+			tar -xf "${url##*/}" 1>&1 2>/dev/null
+			cd ./*/ 2> /dev/null || { error_log 1 "install $PACKAGE"; fancy_message warn "Could not enter into the downloaded archive"; }
```

Comment:
Add cleanup here.

Reply:
Invalid

Reply:
Invalid, since it's not always an achive. We didn't even need to log as error, it's just a warning